controllers/release/adapter.go

```diff
@@ -561,20 +658,49 @@ func (a *adapter) validateAuthor() *controller.ValidationResult {
 
 // validateProcessingResources will ensure that all the resources needed to process the Release exist.
 func (a *adapter) validateProcessingResources() *controller.ValidationResult {
-	resources, err := a.loader.GetProcessingResources(a.ctx, a.client, a.release)
+	releasePlan, err := a.loader.GetReleasePlan(a.ctx, a.client, a.release)
 	if err != nil {
-		if resources == nil || resources.ReleasePlanAdmission == nil || errors.IsNotFound(err) {
+		if errors.IsNotFound(err) {
 			a.release.MarkValidationFailed(err.Error())
 			return &controller.ValidationResult{Valid: false}
 		}
-
 		return &controller.ValidationResult{Err: err}
 	}
+
+	if releasePlan.Spec.Target != "" {
+		resources, err := a.loader.GetProcessingResources(a.ctx, a.client, a.release)
+		if err != nil {
+			if resources == nil || resources.ReleasePlan == nil || resources.ReleasePlanAdmission == nil || errors.IsNotFound(err) {
+				a.release.MarkValidationFailed(err.Error())
+				return &controller.ValidationResult{Valid: false}
+			}
+
+			return &controller.ValidationResult{Err: err}
+		}
+	}
 	return &controller.ValidationResult{Valid: true}
 }
 
-// validatePipelineRef checks that the managed Release PipelineRun ref passes the checks from the ReleaseServiceConfig.
+// validatePipelineRef checks that the Release PipelineRun ref passes the checks from the ReleaseServiceConfig.
 func (a *adapter) validatePipelineRef() *controller.ValidationResult {
+	releasePlan, err := a.loader.GetReleasePlan(a.ctx, a.client, a.release)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			a.release.MarkValidationFailed(err.Error())
+			return &controller.ValidationResult{Valid: false}
+		}
+		return &controller.ValidationResult{Err: err}
+	}
+
+	if releasePlan.Spec.Pipeline != nil {
+		if !a.releaseServiceConfig.Spec.Debug && releasePlan.Spec.Pipeline.PipelineRef.IsClusterScoped() {
```

Comment:
This is duplicated code that could and should go into a function.

Reply:
fixed.